ih-core/bin/ih-setup

```diff
@@ -0,0 +1,342 @@
+#!/bin/bash
+
+export IH_CORE_DIR
+IH_CORE_DIR=$(dirname "$(dirname "$(realpath "$0")")")
+export IH_CORE_BIN_DIR="$IH_CORE_DIR/bin"
+export IH_CORE_LIB_DIR="$IH_CORE_DIR/lib"
+
+# shellcheck source=../lib/confirm.sh
+source "$IH_CORE_LIB_DIR/confirm.sh"
+# shellcheck source=../lib/colors.sh
+source "$IH_CORE_LIB_DIR/colors.sh"
+# shellcheck source=../lib/log.sh
+source "$IH_CORE_LIB_DIR/log.sh"
+
+# Attempts to return help for the step passed as an argument
+function ih::private::step-help() {
+    if typeset -f "ih::setup::$1::help" >/dev/null; then
```

Comment:
I wasn't familiar with `typeset` and its description sounds like this isn't the most appropriate use for it... maybe use `command -v` to check if a function/executable exists?

Could also be behind a function with a nicer name,, e.g. `if comand_exists "ih::setup::$1::help"; ...`

Reply:
I was using `typeset` because for some reason I thought `command` wouldn't see functions. It does though.

Reply:
Your comment reminded me that I started refactoring all the step-* functions to use a consistent pattern, so I finished that refactoring.